x = 10

def my_function():
    x = x + 5
    print(x)

my_function()

# throws error on line 4? x hasn't been assigned yet in local scope